roofit/roofitcore/src/TestStatistics/RooUnbinnedL.cxx

```diff
@@ -0,0 +1,185 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+/**
+\file RooUnbinnedL.cxx
+\class RooUnbinnedL
+\ingroup Roofitcore
+
+Class RooUnbinnedL implements a a -log(likelihood) calculation from a dataset
+and a PDF. The NLL is calculated as
+<pre>
+ Sum[data] -log( pdf(x_data) )
+</pre>
+In extended mode, a (Nexpect - Nobserved*log(NExpected) term is added
+**/
+
+#include <TestStatistics/RooUnbinnedL.h>
+
+#include "RooAbsData.h"
+#include "RooAbsPdf.h"
+#include "RooAbsDataStore.h"
+
+namespace RooFit {
+namespace TestStatistics {
+
+RooUnbinnedL::RooUnbinnedL(RooAbsPdf *pdf, RooAbsData *data,
+                           RooAbsL::Extended extended)
+   : RooAbsL(RooAbsL::ClonePdfData{pdf, data}, data->numEntries(), 1, extended)
+{}
+
+RooUnbinnedL::RooUnbinnedL(const RooUnbinnedL &other)
+   : RooAbsL(other), apply_weight_squared(other.apply_weight_squared), _first(other._first)
+{}
+
+
+bool RooUnbinnedL::processEmptyDataSets() const
+{
+   return extended_;
```

Comment:
inline

Reply:
Same as in RooSumL, this function is no longer used.